From 8c5a4e5c46d04f7153e0b0ff9010ffb8d5068460 Mon Sep 17 00:00:00 2001
From: Ray Essick <essick@google.com>
Date: Mon, 6 Dec 2021 10:22:33 -0800
Subject: [PATCH] Safetynet logging for b/204445255

Bug: 204445255
Test: poc from original bug
Change-Id: I569477d0771e1c03318df9ef271cf3201d472c99
(cherry picked from commit 94e58d6b2497d2e0f7e86e2c979e7f6958c84590)
Merged-In:I569477d0771e1c03318df9ef271cf3201d472c99
---
 media/libmediametrics/include/media/MediaMetricsItem.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/media/libmediametrics/include/media/MediaMetricsItem.h b/media/libmediametrics/include/media/MediaMetricsItem.h
index e36f0a0f3..7ce75ed4f 100644
--- a/media/libmediametrics/include/media/MediaMetricsItem.h
+++ b/media/libmediametrics/include/media/MediaMetricsItem.h
@@ -27,6 +27,7 @@
 #include <variant>
 
 #include <binder/Parcel.h>
+#include <log/log.h>
 #include <utils/Errors.h>
 #include <utils/Timers.h> // nsecs_t
 
@@ -469,6 +470,7 @@ protected:
         do {
             if (ptr >= bufferptrmax) {
                 ALOGE("%s: buffer exceeded", __func__);
+                android_errorWriteLog(0x534e4554, "204445255");
                 return BAD_VALUE;
             }
         } while (*ptr++ != 0);
-- 
2.35.0.263.gb82422642f-goog

